C7Engine/EntryPoints/CityInteractions.cs

```diff
@@ -21,5 +21,16 @@ public static void BuildCity(int x, int y, string playerGuid, string name)
 			tileWithNewCity.cityAtTile = newCity;
 			tileWithNewCity.overlays.road = true;
 		}
+
+		public static void DestroyCity(int x, int y) {
+			Tile tile = EngineStorage.gameData.map.tileAt(x, y);
+			tile.DisbandNonDefendingUnits();
+			while (tile.cityAtTile.size > 0) {
+				tile.cityAtTile.RemoveCitizen();
+			}
+			tile.cityAtTile.owner.cities.Remove(tile.cityAtTile);
+			EngineStorage.gameData.cities.Remove(tile.cityAtTile);
+			tile.cityAtTile = null;
```

Comment:
It seems like a code smell to do each of these separately at this level, but I'm not quite sure how it should be done. I suppose the component model will play a role in that, since there would be a component for handling city data.

Reply:
Yeah, the thing is there are multiple places that reference the city object so it is convenient to reference it from various viewpoints, but that means they all have to be cleaned up too.  On the balance, I'd rather have to clean up a few reference but have those different views than to only be able to access cities via `EngineStorage.gameData.cities`.

My intention with this method is that it will be a one-stop-shop for city destruction, so e.g. it can be re-used when we add the right-click "Abandon city" action.  It may move and it may be modified, but by having one place that handles everything, the fact that there are a few references to clean up shouldn't result in forgetting to clean them up.